video/out/opengl/context.c

```diff
@@ -295,6 +297,7 @@ bool ra_gl_ctx_submit_frame(struct ra_swapchain *sw, const struct vo_frame *fram
     case FLUSH_YES:
         gl->Flush();
     }
+#endif
```

Comment:
I'd just set `gl->Flush` to a dummy function instead. Does it still cause issues with opengl-cb?

Reply:
i didn't have any issues anymore with it but @AirPort still had some issues (https://github.com/mpv-player/mpv/issues/4615#issuecomment-326767362) with his integrated GPU.

wasn't too sure how to handle the overwriting of gl->Flush() since i have to use it [here](https://github.com/mpv-player/mpv/blob/master/video/out/opengl/context_cocoa.c#L161). including the header file for the function leads to some compile warning:
```
../video/out/opengl/gl_headers.h:614:9: warning: 'GL_DRAW_FRAMEBUFFER_BINDING' macro redefined [-Wmacro-redefined]
#define GL_DRAW_FRAMEBUFFER_BINDING       0x8CA6
        ^
/System/Library/Frameworks/OpenGL.framework/Headers/glext.h:360:9: note: previous definition is here
#define GL_DRAW_FRAMEBUFFER_BINDING                          GL_FRAMEBUFFER_BINDING
```

Reply:
Apple's GL headers being retardedly different from other headers again... But they probably still end up at the same constant value and it's just an annoying warning?

Normally, compilers ignore redundant `#define` statements, which is why our GL headers normally don't clash with system ones. But here Apple decided to define a constant to some other define (because they have the same value or so), which is different from what we do and thus emits a warning.

Can you just avoid including any system GL headers? Otherwise you could add an exception with an ifdef here.

Reply:
Regarding glFlush, you could just store the real function pointer to it in the priv struct.

Reply:
i thought about storing a function pointer in the priv struct too. though i was sure this is was to dirty to do. anyway i pushed the change for this. bloody swift won't let me get function pointers.